core/src/main/java/tc/oc/pgm/stats/menu/items/TeamStatsMenuItem.java

```diff
@@ -40,43 +42,28 @@ public class TeamStatsMenuItem implements MenuItem {
 
   private final NamedTextColor RESET = NamedTextColor.GRAY;
 
-  public TeamStatsMenuItem(
-      Match match,
-      Competitor team,
-      Collection<MatchPlayer> relevantObservers,
-      Collection<UUID> relevantOfflinePlayers) {
-    StatsMatchModule smm = match.needModule(StatsMatchModule.class);
+  public TeamStatsMenuItem(Match match, Competitor team, Map<UUID, PlayerStats> playerStats) {
 
     this.team = team;
     this.members = Lists.newArrayList();
-    this.stats = new TeamStats(team, smm);
+    this.stats = new TeamStats(team, new ArrayList<>(playerStats.values()));
```

Comment:
i'd avoid the relocation, just pass accept a Collection<PlayerStats> in the constructor instead
```suggestion
    this.stats = new TeamStats(team, playerStats.values());
```